src/ipcapi.ts

```diff
@@ -250,4 +254,50 @@ export function initIpcApi(ipcMain: IpcMain) {
 	ipcMain.on("shape", syncwrap((e, wnd: BigInt, rects: Rectangle[]) => {
 		native.setWindowShape(wnd, rects);
 	}));
+
+	ipcMain.handle("openapp", async (e, url) => {
+		try {
```

Comment:
`.handle` and `.on` are very different in that `.handle` is async in nature and also properly catches and rethrows errors in the sender. `.on` is synchronous and electron doesn't rethrow errors, that's why i wrap those calls in `{error:..., success:...}` (this is only used in legacy api's that have to be sync)

In this case all the try catch stuff can simply be removed.